Introduction To Network Analysis Using Wireshark (#1314)
section-engineering-education/engineering-education: articles/introduction-to-wireshark/index.md

### Introduction to Network Analysis Using Wireshark
As an IT professional, one of the most powerful networking tools you will find yourself using is ```Wireshark```, a tool mostly used in the analysis of network packets.  
In this tutorial, we will be looking at an overview of ```Wireshark```, how it works, and several other uses. 

### Requirements
* Networking Basics
* TCP/IP stack
* Reading and interpreting packet headers
* Routing and port forwarding
* DHCP

### What is Wireshark?

Wireshark software's history goes back to 1998, as an open-source project started by volunteer network experts around the world.  
The main aim was to build a standard network protocol analyzing tool.  

> Therefore, ```Wireshark``` is simply a network packet analyzer that is mainly used to capture the data on your network. This data is then presented in a form understandable by human beings.  

Reviewer: ```suggestion
_Wireshark_ is a network packet analyzer that is used to capture the data on your network. This data is presented in a human-understandable form.  
```

> In case you find yourself in a position where you need to try and troubleshoot every single network packet, ``` Wireshark``` is the right tool for you!

Reviewer: ```suggestion
In case you find yourself in a position where you need to try and troubleshoot network packets, _Wireshark_ is the right tool for you!
```

> This is a very powerful tool, as it enables you, the user, to perform various operations such as:

Reviewer: ```suggestion
This is a powerful tool that enables you to perform various operations such as:
```

* Troubleshooting your network.
* Performing security operations that are used to detect security threats such as port scanning.  
* Learning more about your network protocols at the microscopic level.
* Performing analysis of voice over the internet(VoIP).

### How Wireshark works

> As we have said, ```Wireshark``` is a packet analyzer, sometimes other people refer to it as a packet sniffer, so do not get confused.  

Reviewer: Replace all ```Wireshark``` with _Wireshark_

> How it works is very simple, let's take your machine, for example, connected to your ```WIFI``` or the network you're currently using, and you want to determine what is currently going on, on your network.  

Reviewer: Don't use variable formatter for non-variables. Use italics or bold for keywords and code formatting options for variables and code snippets.

Reviewer: ```WIFI``` -> _WiFi_

> What happens is that whenever you use ```Wireshark```, it will capture your network traffic(this refers to the data moving currently on your network) and record every single data offline.  

Reviewer: ```suggestion
When you use _Wireshark_, it captures your network traffic (the data moving currently on your network) and records the movement of data offline.
```

> What happens is that whenever you use ```Wireshark```, it will capture your network traffic(this refers to the data moving currently on your network) and record every single data offline.  

Reviewer: Don't use future tense in your writing. Use present tense. Avoid it will capture. Instead, write it captures.

License GPLv2+: GNU GPL version 2 or later <https://www.gnu.org/licenses/gpl-2.0.html>
This is free software; see the source for copying conditions. There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiled (64-bit) with Qt 5.12.8, with libpcap, with POSIX capabilities (Linux),
with libnl 3, with GLib 2.64.2, with zlib 1.2.11, with SMI 0.4.8, with c-ares
1.15.0, with Lua 5.2.4, with GnuTLS 3.6.13 and PKCS #11 support, with Gcrypt
1.8.5, with MIT Kerberos, with MaxMind DB resolver, with nghttp2 1.40.0, with
brotli, with LZ4, with Zstandard, with Snappy, with libxml2 2.9.10, with
QtMultimedia, without automatic updates, with SpeexDSP (using system library),
with SBC, with SpanDSP, without bcg729.

Running on Linux 5.4.0-58-generic, with          Intel(R) Celeron(R) CPU B830 @
1.80GHz (with SSE4.2), with 3824 MB of physical memory, with locale en_US.UTF-8,
with libpcap version 1.9.1 (with TPACKET_V3), with GnuTLS 3.6.13, with Gcrypt
1.8.5, with brotli 1.0.7, with zlib 1.2.11, binary plugins supported (0 loaded).

Built using gcc 9.3.0.
jumamiller@OpijaKaeli:~$ 

```
#### Step 5: Launch Wireshark

> To start Wireshark, run the following command on the terminal:- 

Reviewer: command in the terminal.

```console
wireshark
```
```console
jumamiller@OpijaKaeli:~$ wireshark
|
```
By running this command will redirect you to the Wireshark software window.

#### Fig 1.2 Wireshark Window
![Wireshark-window](/engineering-education/introduction-to-wireshark/wireshark-window.jpg) .  

Congratulations, you now have ```Wireshark``` installed in your system and running.

### Your first packet capture
We have Wireshark in our system installed. Let's dive in and start experimenting with its features.  

#### 1. Wireshark Graphical User Interface
From ```Fig 1.2``` above, Wireshark contains some of the commonly used menus,

> ``` File, Edit, View, Go, Capture, Analyze, Statistics, Telephony, Wireless, Tools, and Help```.  

Reviewer: ```suggestion
_File, Edit, View, Go, Capture, Analyze, Statistics, Telephony, Wireless, Tools, and Help_
```

Normally, when you start a Wireshark without opening a capture file or starting a capture process, a welcome screen is displayed.  
This window will always display currently opened capture files and the capture available interfaces.  

Therefore the first step involves selecting the network interface to capture its data.  
Remember, the interfaces are different for different operating systems.  

#### Fig 1.3 Network Interfaces
![Network-Interfaces](/engineering-education/introduction-to-wireshark/wireshark-network-interface.jpg)

From the above screenshot, we have 6 interfaces, which we can then choose from.  

Hit on the ```Capture``` button just below the ``` Welcome to Wireshark```.  
This will prompt you to another window, as seen below.  

#### Fig 1.4 Network Interfaces 2
![Network-Interfaces](/engineering-education/introduction-to-wireshark/wireshark-network-interfaces-2.jpg)

Now select any of the interfaces you're presented with. In this example, we're going to explore the UDP Listener option.  
Then click on the start button in the far left corner.  
At this stage, you should be able to get an output like the one presented below.  

> #### Fig 1.5 Network Traffics

Reviewer: Network Traffic

![Network-Traffics](/engineering-education/introduction-to-wireshark/wireshark-live-capture.jpg)

In the capture panel above, you're able to notice that network packet capturing is in sequential order, with each line representing each packet captured.  

The information is displayed in tabular form, with rows and columns.  
Each row represents the packet collected, while columns provide you with additional information such as time, protocols, length, et cetera. You can refer to the screenshot above.  

Let's have a look at these columns and what type of information they provide us with.  
* ```No``` - Represents a unique network packet sequence number. This can be used to identify a given packet.  

> * ``` Time ```- This is the time at which a particular packet was captured.  

Reviewer: \``` -> while using this, there should be no space after the formatter. Be careful as it can lead to formatting errors.

* ```Source ```- This represents where we are getting the packets from. This is denoted as Internet Protocols(IP Addresses).
* ``` Destination```- This is used to represent the Internet Protocol(IP Address) where the packet is going.  
* ```Protocol```- This refers to the protocol of the data you have captured. This could be ```TCP, ARP et cetera```
* ```Length```- This is used to represent the size of the packet captured.
* ``` Info```- This gives you additional information about the packet you have captured.  

NOTE:- Each protocol is represented with its color scheme. In our case, for example, the ```TCP ``` protocol has a ```#cccccc``` background.  This helps the user to easily differentiate between these protocols.  

Congratulations, you have been able to capture network packets, and that's how simple Wireshark is user friendly. 

### Packet Details Panel:-
Now that we can capture some data try to click on a single row, and you will notice that some data are being displayed on the immediate window.  

> #### Fig 1.5 Packet Details

Reviewer: Packet details

![Single-Packet-Details](/engineering-education/introduction-to-wireshark/wireshark-single-packet-details.jpg)
On the highlighted protocol, click on it to get more details about this SSDP protocol as shown below:-  

> #### Fig 1.6 More on packet Details

Reviewer: More on packet details

![Protocol-details](/engineering-education/introduction-to-wireshark/wireshark-protocol-details.jpg)

The information above tells us more about the packet captured, including the device used. This data is very important, especially in system hacks, where they can be collected for forensics.  

> ### Packet Bytes Panel

Reviewer: Packet bytes panel

Do this for all headings. Only the first word should begin with a capital letter.
